pkg/datagatherer/k8s/generic_test.go

```diff
@@ -60,14 +61,39 @@ func asUnstructuredList(items ...*unstructured.Unstructured) *unstructured.Unstr
 	}
 }
 
+func TestNewDataGathererWithClient(t *testing.T) {
+	config := Config{
+		IncludeNamespaces:    []string{"a"},
+		GroupVersionResource: schema.GroupVersionResource{Group: "foobar", Version: "v1", Resource: "foos"},
+	}
+	cl := fake.NewSimpleDynamicClient(runtime.NewScheme())
+	dg, err := config.newDataGathererWithClient(cl)
+
+	if err != nil {
+		t.Errorf("expected no error but got: %v", err)
+	}
+
+	expected := &DataGatherer{
+		cl:                   cl,
+		groupVersionResource: config.GroupVersionResource,
+		// it's important that the namespaces are set as the IncludeNamespaces
+		// during initialization
+		namespaces: config.IncludeNamespaces,
```

Comment:
I'm not sure this test is a good idea. It feels a bit much tbh.

However, it's the only place we validate that the include-namespaces make it to the dg configuration...

I thought it worth adding, but I'm indifferent as to it staying. `newDataGathererWithClient` isn't ideal.